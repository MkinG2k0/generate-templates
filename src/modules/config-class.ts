import { IConfig } from '../index.js'
import { Arg, DataArg } from './arg-class.js'
import { log } from './logs-class.js'
import { Parser } from './parser-class.js'
import { Reader } from './reader-class.js'

interface TConfig {}

export class Config implements TConfig {
  pathRun = ''
  generateConfig?: IConfig
  args?: DataArg = undefined

  constructor(process: NodeJS.Process) {
    this.args = new Arg(process.argv).data
    this.pathRun = process.cwd()
  }

  async read() {
    if (!this.args?.pathConfig) {
      log.error('Choose path to config').view(this.args)
      return
    }
    const reader = new Reader(this.args.pathConfig, { path: this.pathRun })
    const data = await reader.read()

    if (data) {
      this.generateConfig = (await Parser.parse(data)) as IConfig
      log.setDebug(log.debug || this.generateConfig.debug)
    }

    return this.generateConfig
  }
}
