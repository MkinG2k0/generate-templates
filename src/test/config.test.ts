import { describe, expect } from 'vitest'
import { Config } from '../modules/config-class.js'
import { processFailedMock, processMock } from './mock/process.js'

describe('Config class', async () => {
  test('Config success', async () => {
    const config = new Config(processMock)
    const dataConfig = await config.read()
    expect(dataConfig).matchSnapshot()
  })

  test('Config error path', async () => {
    const config = new Config(processFailedMock)
    const dataConfig = await config.read()
    expect(dataConfig).toBeUndefined()
  })
})
