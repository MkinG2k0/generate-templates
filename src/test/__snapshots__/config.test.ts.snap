// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Config class > Config success 1`] = `
{
  "debug": true,
  "replaceDataName": "fileName",
  "replaceFileName": "{name}",
  "replaceNameExt": [
    [
      "FileName.txt",
      "tsx",
    ],
  ],
  "templates": {
    "comp": {
      "generate": "generated/Components",
      "template": "templates/comp",
    },
    "page": {
      "generate": "generated/Pages/",
      "replaceDataName": /FileName/g,
      "template": "templates/page",
    },
  },
}
`;
